test/test_runtime.py

```diff
@@ -444,6 +447,68 @@ def test_require_collection(runtime_tmp: Runtime) -> None:
     runtime_tmp.require_collection("community.molecule", "0.1.0")
 
 
+@dataclass
+class ScanSysPath:
+    """Parameters for scan tests."""
+
+    isolated: bool
+    scan: bool
+    expected: bool
+
+    def __str__(self) -> str:
+        """Return a string representation of the object."""
+        parts = [
+            f"{field.name}{str(getattr(self, field.name))[0]}" for field in fields(self)
+        ]
+        return "-".join(parts)
+
+
+@pytest.mark.parametrize(
+    ("param"),
+    (
+        ScanSysPath(isolated=True, scan=True, expected=False),
+        ScanSysPath(isolated=True, scan=False, expected=False),
+        ScanSysPath(isolated=False, scan=True, expected=True),
+        ScanSysPath(isolated=False, scan=False, expected=False),
+    ),
+    ids=str,
+)
+def test_scan_sys_path(
+    monkeypatch: MonkeyPatch,
+    tmp_path: Path,
+    runtime_tmp: Runtime,
+    param: ScanSysPath,
+) -> None:
+    """Confirm sys path is scanned for collections.
+
+    :param monkeypatch: Fixture for monkeypatching
+    :param tmp_path: Fixture for a temp directory
+    :param runtime_tmp: Fixture for a Runtime object
+    :param param: The parameters for the test
+    """
+    runtime_tmp.install_collection(
+        V2_COLLECTION_TARBALL,
+        destination=tmp_path,
+    )
+    runtime_tmp.config.collections_paths.remove(str(tmp_path))
+
+    # Set the runtime to the test parameters
+    runtime_tmp.isolated = param.isolated
+    runtime_tmp.config.collections_scan_sys_path = param.scan
+    monkeypatch.syspath_prepend(str(tmp_path))
+    runtime_tmp._add_sys_path_to_collection_paths()
+
+    try:
+        runtime_tmp.require_collection(
+            name="community.molecule",
+            version="0.1.0",
+            install=False,
+        )
+    except InvalidPrerequisiteError:
```

Comment:
if there is no exception, the param.expected will not be used.
The parameterized is not precise

Reply:
ha, good catch, TY